### PR TITLE
Fix Caves

### LINKED ISSUE #87: Caves Don't Spawn
## Description
***lmfao*** i forgot to fix caves

## To Reproduce

1. go in fraiya dimension
2. go in spectator mode
3. go underground
4. cry

## Expected Behavior
caves should exist

## Screenshots/Logs
wtf

## Specifications
Operating System | Minecraft Version | Mod Version
---------------- | ----------------- | -----------
Linux Ubuntu 20.04 | 1.17 | 2-SNAPSHOT

### PR DESCRIPTION
**Pull request by sylv256**
## Changelog
 - fix caves
 - make Nebulaes Bane caves cooler and update them because mojank decided to pull our old cave code from under ~~*among*~~ us

## Todo
 - [ ] Tweaked
 - [ ] Debugged

Fixes #87

### CODE REVIEW

**Comment by valoeghese** [comment]
I have plans for caves idk if I made an issue or comment for it if not let me know and i’ll share

**Comment by valoeghese** [comment]
#57 is related
and I found a disocrd message from February that basically says
use noise to define about 1/3 as crystal jungle gen, in which there are large stalactites and stalagmites, and literal plants and vines and trees made of crystal. I think the crystal “plants” were gonna gen similarly to end chorus fruits, but some would be upside down. Could write it myself

File: ensure-java-16 [hunk 1]
```diff
@@ -0,0 +1,21 @@
+#!/bin/bash
+
+# https://github.com/jitpack/jitpack.io/issues/4355#issuecomment-744678253
```
Comment (valoeghese) [inline ensure-java-16]: ah yes fixing jutpack

File: src/main/java/me/hydos/lint/world/carver/LintConfiguredCarvers.java [hunk 2]
```diff
@@ -19,7 +19,31 @@
...
+import net.minecraft.world.gen.carver.CarverDebugConfig;
+import net.minecraft.world.gen.carver.ConfiguredCarver;
+import net.minecraft.world.gen.heightprovider.BiasedToBottomHeightProvider;
+
+// FIXME: why won't it work irr (this might have something to do with the biomes but i'm not sure)
 public interface LintConfiguredCarvers {
-//	ConfiguredCarver<CarverConfig> CAVE = Registry.register(BuiltinRegistries.CONFIGURED_CARVER, Lint.id("nebulaes_bane"), LintCaveCarver.INSTANCE.configure(new ProbabilityConfig(0.09F)));
+	ConfiguredCarver<LintCaveCarverConfig> NEBULAES_BANE = register("nebulaes_bane",
+					new LintCaveCarverConfig(
+							/*0.09F*/0.33333334F,
```
Comment (valoeghese) [inline src/main/java/me/hydos/lint/world/carver/LintConfiguredCarvers.java]: what does this change do, just make caves rarer?

Comment (valoeghese) [reply src/main/java/me/hydos/lint/world/carver/LintConfiguredCarvers.java]: Please describe what each of your changes do

File: src/main/java/me/hydos/lint/world/gen/terrain/TerrainChunkGenerator.java [hunk 3]
```diff
@@ -303,7 +303,7 @@ public void buildSurface(ChunkRegion region, Chunk chunk) {
 				region.getBiome(mutable.set(startX + xo, height, startZ + zo)).buildSurface(rand, chunk, x, z, height, noise,
 						LintBlocks.FUSED_STONE.getDefaultState(), Blocks.WATER.getDefaultState(), this.getSeaLevel(), region.getBottomY(), region.getSeed());
 
-				// bedrock
+				// bed rock :crushed:
```
Comment (valoeghese) [inline src/main/java/me/hydos/lint/world/gen/terrain/TerrainChunkGenerator.java]: no

File: src/main/resources/mixins.lint.json [hunk 4]
```diff
@@ -2,7 +2,7 @@
   "required": true,
   "minVersion": "0.8",
   "package": "me.hydos.lint.mixin",
-  "compatibilityLevel": "JAVA_8",
+  "compatibilityLevel": "JAVA_16",
```
Comment (valoeghese) [inline src/main/resources/mixins.lint.json]: don't think mixin actualyl supports j16

File: src/main/java/me/hydos/lint/world/biome/Biomes.java [hunk 5]
```diff
@@ -130,7 +131,8 @@
 			.spawnSettings(DEFAULT_SPAWN_SETTINGS.build())
 			.generationSettings(new GenerationSettings.Builder()
 					.surfaceBuilder(MF_SB)
-					.carver(GenerationStep.Carver.AIR, ConfiguredCarvers.CAVE)
+					.carver(GenerationStep.Carver.AIR, LintConfiguredCarvers.NEBULAES_BANE)
+					.carver(GenerationStep.Carver.AIR, ConfiguredCarvers.PROTOTYPE_CAVE)
```
Comment (valoeghese) [inline src/main/java/me/hydos/lint/world/biome/Biomes.java]: what is prototype_cave

Comment (valoeghese) [reply src/main/java/me/hydos/lint/world/biome/Biomes.java]: is that the underground squashed spheres / circular rooms

**Comment by valoeghese** [comment]
I'll do the actual fancy cave stuff when the cool cave shape pr is done, tested, and tehc lists what the actual changes are because I can't be bothered to read every change line and figure out what caves are doing different

**Comment by valoeghese** [comment]
Tehc Support Guy please fix

**Comment by sylv256** [comment]
sorry i'll pr soon i'm very busy

**Comment by valoeghese** [comment]
Tehc I see u have caves working in the discord. when pr?

**Comment by sylv256** [comment]
> Tehc I see u have caves working in the discord. when pr?

#87

**Comment by sylv256** [comment]
#100 for new pr

**Pull request closed without merging**